app/policies/adopter_application_policy.rb

```diff
@@ -0,0 +1,44 @@
+class AdopterApplicationPolicy < ApplicationPolicy
+  authorize :pet, optional: true
```

Comment:
Do we need to change the naming of adopter_profile to adopter_foster_profile here?

Reply:
I think I will update this to just `verify_profile!` instead to decouple the method from the account type naming conventions. Any objections to that?